tests/parser/array-destructuring.ts.fixture.ts

```diff
@@ -0,0 +1,5 @@
+const arr = [1, 2];
+const a = arr[0], b = arr[1];
+const x = [3, 4][0], y = [3, 4][1];
```

Comment:
Would love to hear some suggestion how to implement a "dummy" variable to avoid the double invocation

Reply:
We typically compile the `Statement`s directly, i.e. not trying to "transpile" on the parser level, so we not only get a 1:1 roundtrip when serializing the AST again, but also do not have to deal with temporary AST nodes like the one that would be necessary here. Here, the compiler would typically implement a `compileDestructuringAssignment` method directly compiling to Binaryen IR, which may then use a temporary Wasm local.

Reply:
Make sense. From what I hear I need to set the `NodeKind` to `DESTRUCTED_VARIABLE`. And then handle it in `compileStatement()` switch?

Reply:
Or add a flag to the statement and keep the type to `VARIABLE`